chaosgcp/apphub/actions.py

```diff
@@ -0,0 +1,315 @@
+# Copyright 2024 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Dict, Any, List, Generator, Optional
+
+from google.cloud import compute_v1, apphub_v1
+from google.cloud import resourcemanager_v3
+
+from chaoslib.types import Configuration, Secrets
+from chaosgcp.lb.actions import (
+    inject_traffic_faults,
+    remove_fault_injection_traffic_policy,
+)
+from chaosgcp import get_context, load_credentials
+from chaoslib.exceptions import ActivityFailed
+
+
+__all__ = [
+    "inject_fault_if_url_map_exists_app_hub",
+    "remove_fault_if_url_map_exists_app_hub",
+]
+logger = logging.getLogger("chaostoolkit")
+
+
+def apphub_app_list_services(
+    host_project_id: str,
+    project_id: str,
+    location: str,
+    application_id: str,
+    configuration: Configuration = None,
+    secrets: Secrets = None,
+) -> List[str]:
+    """Retrieves a list of service URIs associated with an AppHub application.
+
+    This function queries the AppHub API to list all services deployed within
+    a specific application and extracts their associated URIs.
+
+    Args:
+        host_project_id (str): The ID of the Host Apphub project.
+        project_id (str): The ID of the Google Cloud project to inject fault into.
+        location (str): The geographic location (e.g., "us-central1") of the application.
+        application_id (str): The unique identifier for the AppHub application.
+
+    Returns:
+        list: A list of strings, where each string is the URI of a service associated
+              with the specified application. The URIs are formatted to contain the
+                      project ID instead of the project number only for project_id."""
+    try:
+        name_app = f"projects/{host_project_id}/locations/{location}/applications/{application_id}"
+        credentials = load_credentials(secrets)
+        client = apphub_v1.AppHubClient()
+        request = apphub_v1.ListServicesRequest(parent=name_app)
+        service_uris = []
+        updated_uri = None
+        logger.info("Getting attached Services of given Application in Apphub")
```

Comment:
can we make this a debug level?

Reply:
It should be visible in info,otherwise there will be same info with , without apphub application. It is really required to distinguish , and show apphub related things .